SteamKit2/SteamKit2/Types/KeyValue.cs

```diff
@@ -720,7 +756,8 @@ public bool ReadAsBinary( Stream input )
                 this.Children.Add( current );
             }
 
-            return input.Position == input.Length;
+            // it's okay if we haven't reached the end of the input stream
+            return input.Position <= input.Length;

```

Comment:
I believe some streams can seek past their length, but I'm not sure what kind of behavior we'll get if we attempt to read a truncated KV, for example.